src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php

```diff
@@ -27,11 +27,15 @@ class CollectionType extends AbstractType
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         if ($options['allow_add'] && $options['prototype']) {
-            $prototype = $builder->create($options['prototype_name'], $options['entry_type'], array_replace(array(
+            $prototypeOptions = array_replace(array(
                 'label' => $options['prototype_name'].'label__',
-            ), $options['entry_options'], array(
-                'data' => $options['prototype_data'],
-            )));
+            ), $options['options']);
+
+            if ($options['prototype_data'] !== null) {
```

Comment:
Symfony CS prefers to use Yoda conditions: `if (null !== $options['prototype_data']) {`